R/eval_forecasts.R

```diff
@@ -70,7 +69,15 @@
 #'   If you want to score the median (i.e. \code{range = 0}), you still
 #'   need to include a lower and an upper estimate, so the median has to
 #'   appear twice.}
-#' @param summarised if \code{TRUE}, only one average score is returned per model
+#' @param by character vector of columns to group scoring by. The default
+#' is \code{c("model")}, but you could e.g. group over different locations
+#' or horizons. Note that a column of the corresponding name must be
+#' present in the data. If you don't want any grouping, set \code{by = NULL}
+#' @param summarised if \code{TRUE}, only one average score is returned per
+#' grouped unit
+#' @param ... pass down additional arguments to lower-level functions. One
```

Comment:
not sure I really like the dual nature of this - perhaps have a pit_params and interval_params argument that can be used as a list and called with do.call on the relevant function. Current implementation really opens you up to some weird bugs,

Reply:
sounds good as well. I'm not yet sure how to actually implement it. So my suggestion would be to merge as is and then either you can make a quick PR or I'll revisit it later. The only thing I ever used do.call for was do.call(rbind) :D

Reply:
do.call(function, list_of_function_arguments_that_are_named_as_in_the_function)

Reply:
thanks, I'll have a go :)

Reply:
now updated. can you check if that looks good? thanks

Reply:
looks good to me.